com.regnosys.rosetta.lib/src/main/java/com/rosetta/model/lib/validation/ValidationResult.java

```diff
@@ -82,6 +82,9 @@ public String getDefinition() {
 		
 		@Override
 		public Optional<String> getFailureReason() {
+			if (!failureReason.isEmpty() && modelObjectName.endsWith("Report")) {
+				return getUpdatedFailureReason();
+			}
```

Comment:
This change will again invalidate the expected property
```
var reason = Optional.of("The validation failed");
var validationResult = new ModelValidationResult(..., reason);
assertEquals(reason, validationResult.getFailureReason());
```
... but okay, I'll approve to get things rolling for you